theano/configdefaults.py

```diff
@@ -1254,11 +1254,16 @@ def check_mkl_openmp():
     try:
         import mkl
         if '2018' in mkl.get_version_string():
-            raise RuntimeError('To use MKL 2018 with Theano you MUST set "MKL_THREADING_LAYER=GNU" in your environment.')
+            raise RuntimeError("""
+To use MKL 2018 with Theano either update the numpy conda packages to
+their latest build or set "MKL_THREADING_LAYER=GNU" in your
+environment.
+""")
     except ImportError:
         raise RuntimeError("""
-Could not import 'mkl'.  Either install mkl-service with conda or set
-MKL_THREADING_LAYER=GNU in your environment for MKL 2018.
+Could not import 'mkl'.  If you are using conda, update the numpy
+packages to the latest build otherwise, set MKL_THREADING_LAYER=GNU in
```

Comment:
I'm not sure updating numpy is the right solution here, especially if they have the non-mkl version.

Reply:
We only call this function if we attempt to link with MKL.  So the non-mkl case is not an issue.

Reply:
Also, if they have the mkl version and update the package to a version with the _mklinit module, then we don't care if they have mkl-service installed or not (at least from this point of view).

Reply:
I agree with @abergeron answer, so I'll approve this PR. But I'll wait for @lamblin to see this before merging.